packages/langium/src/parser/indentation-aware.ts

```diff
@@ -321,19 +312,20 @@ export class IndentationAwareTokenBuilder<Terminals extends string = string, Key
         }
 
         const numberOfDedents = this.indentationStack.length - matchIndentIndex - 1;
+        const newlinesBeforeDedent = text.substring(0, offset).match(/[\r\n]+$/)?.[0].length ?? 1;
```

Comment:
I think this should not only backtrack through all the newline characters, but also whitespace. For example, if we dedent later:

```
{
    value
    <-- whitespace
    <-- whitespace
} <-- Dedent token appears at start of this line
```

This seems pretty weird, as if we remove the whitespace, the dedent already appears in the 3rd line. Since this PR is there to fix behavior such as the folding, I would've thought it should keep the `DEDENT` token in the 3rd line.

Reply:
But if emit the `DEDENT` token on the beginning of the 3rd line, then it means that the whitespace after it will be detected as `INDENT`, and naturally followed by another `DEDENT` on the next line (and once more, for this example). These extra `INDENT`/`DEDENT` tokens are unexpected and will cause parsing errors.

I generally think that trailing whitespace is problematic and should be marked as a parsing error, especially in indentation-sensitive languages.

Reply:
I see. I guess that makes sense. Thanks for the explanation 👍